pkg/prometheus/server/operator.go

```diff
@@ -787,10 +787,22 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return fmt.Errorf("failed to reconcile Thanos config secret: %w", err)
 	}
 
-	// Create governing service if it doesn't exist.
+	if err := c.createOrUpdateThanosConfigSecret(ctx, p); err != nil {
+		return fmt.Errorf("failed to reconcile Thanos config secret: %w", err)
+	}
```

Comment:
mistake?
```suggestion
```

Reply:
Might've accidentally smashed the key binding